docs/_docs/02_features/keep-your-terragrunt-architecture-dry.md

```diff
@@ -0,0 +1,345 @@
+---
+layout: collection-browser-doc
+title: Keep your Terragrunt Architecture DRY
+category: features
+categories_url: features
+excerpt: Learn how to use multiple terragrunt configurations to DRY up your architecture.
+tags: ["DRY", "Use cases", "backend"]
+order: 210
+nav_title: Documentation
+nav_title_link: /docs/
+---
+
+## Keep your Terragrunt Architecture DRY
+
+  - [Motivation](#motivation)
+
+  - [Using include to DRY common Terragrunt config](#using-include-to-dry-common-terragrunt-config)
+
+  - [Using read\_terragrunt\_config to DRY parent configurations](#using-read_terragrunt_config-to-dry-parent-configurations)
+
+
+### Motivation
+
+As covered in [Keep your Terraform code DRY]({{site.baseurl}}/docs/features/keep-your-terraform-code-dry) and [Keep your
+remote state configuration DRY]({{site.baseurl}}/docs/features/keep-your-remote-state-configuration-dry), it becomes
+important to define base Terragrunt configuration files that are included in the child config. For example, you might
+have a **root** Terragrunt configuration that defines the remote state and provider configurations:
+
+```hcl
+remote_state {
+  backend = "s3"
+  config = {
+    bucket         = "my-terraform-state"
+    key            = "${path_relative_to_include()}/terraform.tfstate"
+    region         = "us-east-1"
+    encrypt        = true
+    dynamodb_table = "my-lock-table"
+  }
+}
+
+generate "provider" {
+  path = "provider.tf"
+  if_exists = "overwrite_terragrunt"
+  contents = <<EOF
+provider "aws" {
+  assume_role {
+    role_arn = "arn:aws:iam::0123456789:role/terragrunt"
+  }
+}
+EOF
+}
+```
+
+You can then include this in each of your **child** `terragrunt.hcl` files using the `include` block for each
+infrastructure module you need to deploy:
```

Comment:
Should our default recommendation now be to always include a label on `include`? It's probably good to capture intent and it makes it easier to extend into multiple `include` blocks without introducing a new syntax. In the reference docs, we could keep a mention of the old unlabeled `include` block for backwards compatibility, but label it as deprecated.

Reply:
Good idea. Done: 9ecd02333edc6cbac0a5327a54883be9d6b4d47e